### ISSUE TITLE
Compile Order?

### ISSUE THREAD

**Issue by monkey-ldb** [RazorGenerator/RazorGenerator#207]
Hi,

I tried to use the RazorGenerator.MSBuild nuget,
but I have a problem now:
The views in the website project are using code from the website (MVC.Website) project itself.
That means, they have a using inside like @using MVC.Website.XY.
This results in an compile time error, because while compiling the "MVC.Website.dll" with the code and the views, the compiled views require the dll, which is not compiled at this time.
I cannot find the MVC.Website.dll in the bin folder at this time,
and it seems that the compilation wont access the source code to compile the views.

How to resolve this issue?
Can I change the output for the views somehow?
Maybe a different DLL? Or the compile order?

**Comment by CZEMacLeod** [comment]
@monkey-ldb To clarify - you are using 2 projects, one for the views and one as the main website?
If so then this won't really work as if you just had 2 projects A and B and had B depend on A and code in A that is using B it can't compile.
The solution is to create a third project C which has either the utility and concrete implementations that are required by A and B, or has Interfaces used by A and implemented in B and inject them using some sort of DI.
Traditionally you might end up with projects like
Website -> Views -> Models -> DAL
to segregate your code.

**Comment by Jehoel** [comment]
I don't think that should be happening. To confirm, is that a real compile-time error you see in command-line MSBuild output, or just an IntelliSense error entry in Visual Studio' Error List?
